### PR TITLE
Adding the Match parameter to the Probes to check on the response code

### PR DESCRIPTION
**Pull request by VaijanathB**
This is blocking a current customer. I have added the change to existing test case probe. Ran the tests private but failed on deleting subnet which is a known issue.

### CODE REVIEW

File: azurerm/resource_arm_application_gateway.go [hunk 1]
```diff
@@ -457,6 +457,29 @@ func resourceArmApplicationGateway() *schema.Resource {
...
+										Type:     schema.TypeString,
+										Optional: true,
+										Default:  "*",
+									},
+
+									"status_code": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
```
[the remark this answers is not in the document]
Comment (VaijanathB) [reply azurerm/resource_arm_application_gateway.go]: People can specify status codes as ranges also like 200-300.

File: azurerm/resource_arm_application_gateway.go [hunk 2]
```diff
@@ -1608,6 +1646,20 @@ func flattenApplicationGatewayProbes(input *[]network.ApplicationGatewayProbe) [
 				if threshold := props.UnhealthyThreshold; threshold != nil {
 					settings["unhealthy_threshold"] = int(*threshold)
 				}
+
+				if match := props.Match; match != nil {
+
+					matchConfig := map[string]interface{}{
+						"body": *match.Body,
```
[the remark this answers is not in the document]
Comment (VaijanathB) [reply azurerm/resource_arm_application_gateway.go]: Thanks. Fixed it.

File: azurerm/resource_arm_application_gateway.go [hunk 3]
```diff
@@ -1608,6 +1646,20 @@ func flattenApplicationGatewayProbes(input *[]network.ApplicationGatewayProbe) [
...
+					}
+
+					statusCodes := make([]interface{}, 0)
+					for _, status := range *match.StatusCodes {
```
[the remark this answers is not in the document]
Comment (VaijanathB) [reply azurerm/resource_arm_application_gateway.go]: Fixed.

File: azurerm/resource_arm_application_gateway.go [hunk 4]
```diff
@@ -457,6 +457,29 @@ func resourceArmApplicationGateway() *schema.Resource {
 							Type:     schema.TypeInt,
 							Required: true,
 						},
+
+						"match": {
```
[the remark this answers is not in the document]
Comment (VaijanathB) [reply azurerm/resource_arm_application_gateway.go]: Added.

File: azurerm/resource_arm_application_gateway_test.go [hunk 5]
```diff
@@ -477,6 +477,16 @@ resource "azurerm_application_gateway" "test" {
...
     interval            = 300
     unhealthy_threshold = 8
+    match               = {
+      body              = "*"
+      status_code       = [
+        200,
+        203,
+        202,
+      ]
+    }
```
[the remark this answers is not in the document]
Comment (VaijanathB) [reply azurerm/resource_arm_application_gateway_test.go]: I have added a separate test that verifies the presence of Match block in the probe is being set.

**Comment by VaijanathB** [comment]
@tombuildsstuff Thanks for your review. This is the last fix I will make and this was blocking the customer I was working with. Currently they are working around this using powershell and wanted to fix this ASAP. I have addressed all the review comments.

**Comment by tombuildsstuff** [comment]
The Application Gateway tests are intermittent - but this seems fine:

screenshot 2018-06-28 at 10 37 27

**Pull request merged**